openhtf/io/user_input.py

```diff
@@ -84,7 +85,7 @@ def DisplayPrompt(self, message, text_input=False):
       console_prompt = ConsolePrompt(
           message, functools.partial(self.Respond, self.prompt.id))
       console_prompt.start()
-      self._cond.wait(int(FLAGS.prompt_timeout_s))
+      self._cond.wait(timeout)
```

Comment:
if you do all the other changes i mentioned, this line becomes:
self._cond.wait(timeout_s or FLAGS.prompt_timeout_s)

This has the handy side effect of allowing you to pass 'None' explicitly (or just leave it out), causing an indefinite timeout (usually good to be able to do this so you don't have to make up "really big value"s).